src/gennylib/include/gennylib/v1/GlobalRateLimiter.hpp

```diff
@@ -84,6 +87,30 @@ class BaseGlobalRateLimiter {
      * appropriate back-off strategy if this function returns false.
      */
     bool consumeIfWithinRate(const typename ClockT::time_point& now) {
+
+        // Logic for percentile rates.
+        bool curFullSpeed = _fullSpeed.load();
+        if (curFullSpeed) {
+            _burstCount++;
+            auto _nsSincePhase = ClockT::now().time_since_epoch().count() - _lastEmptiedTimeNS;
```

Comment:
1. don't need `_` for local vars, only instance vars--I was reading it as an instance var and totally confused mysefl until I saw the `auto` here.
2. consider name `nsSincePhaseStarted`

Reply:
Good catch, done.